mne/bem.py

```diff
@@ -1885,65 +1902,87 @@ def make_flash_bem(subject, overwrite=False, show=True, subjects_dir=None,
     run_subprocess_env = partial(run_subprocess, env=env,
                                  cwd=tempdir)
 
+    mri_dir = Path(mri_dir)
+    bem_dir = Path(bem_dir)
     if flash_path is None:
-        flash_path = op.join(mri_dir, 'flash', 'parameter_maps')
+        flash_path = mri_dir / 'flash' / 'parameter_maps'
     else:
-        flash_path = op.abspath(flash_path)
+        flash_path = Path(flash_path).resolve()
     subjects_dir = env['SUBJECTS_DIR']
+    flash_path.mkdir(exist_ok=True, parents=True)
 
     logger.info('\nProcessing the flash MRI data to produce BEM meshes with '
                 'the following parameters:\n'
                 'SUBJECTS_DIR = %s\n'
                 'SUBJECT = %s\n'
                 'Result dir = %s\n' % (subjects_dir, subject,
-                                       op.join(bem_dir, 'flash')))
+                                       bem_dir / 'flash'))
     # Step 4 : Register with MPRAGE
-    logger.info("\n---- Registering flash 5 with MPRAGE ----")
-    flash5 = op.join(flash_path, 'flash5.mgz')
-    flash5_reg = op.join(flash_path, 'flash5_reg.mgz')
-    if not op.exists(flash5_reg):
-        if op.exists(op.join(mri_dir, 'T1.mgz')):
-            ref_volume = op.join(mri_dir, 'T1.mgz')
-        else:
-            ref_volume = op.join(mri_dir, 'T1')
-        cmd = ['fsl_rigid_register', '-r', ref_volume, '-i', flash5,
-               '-o', flash5_reg]
+    flash5 = flash_path / 'flash5.mgz'
+    mef05 = mri_dir / 'flash' / 'mef05.mgz'
+    if flash5_img is None and mef05.exists():
+        flash5_img = mef05
+
+    if _path_like(flash5_img):
+        logger.info(f"Copying flash 5 image {flash5_img} to {flash5}")
+        cmd = ['mri_convert', Path(flash5_img).resolve(), flash5]
         run_subprocess_env(cmd)
+    elif flash5_img is None:
+        if not flash5.exists():
+            raise ValueError(f'Flash 5 image cannot be found at {flash5}.')
     else:
-        logger.info("Registered flash 5 image is already there")
+        logger.info(f"Writing flash 5 image at {flash5}")
+        import nibabel as nib
+        nib.save(flash5_img, flash5)
```

Comment:
Could be streamlined as:
```
_validate_type(flash5_img, (None, 'path-like', SpatialImage), 'flash5_img')
if isinstance(flash5_img, SpatialImage):
    ...
elif flash5_img is not None:
    flash5_img = _check_fname(flash5_img, must_exist=True, overwrite='read')
    ...
```
Then you don't need your own `.resolve` and `.exists` etc.